jetty-websocket/websocket-javax-server/src/main/java/org/eclipse/jetty/websocket/javax/server/config/JavaxWebSocketServletContainerInitializer.java

```diff
@@ -96,28 +107,7 @@ public static void configure(ServletContextHandler context, Configurator configu
     {
         if (!context.isStopped())
             throw new IllegalStateException("configure should be called before starting");
-
-        // In this embedded-jetty usage, allow ServletContext.addListener() to
-        // add other ServletContextListeners (such as the ContextDestroyListener) after
-        // the initialization phase is over. (important for this SCI to function)
-        context.getServletContext().setExtendedListenerTypes(true);
-
-        context.addEventListener(ContainerInitializer.asContextListener(new JavaxWebSocketServletContainerInitializer())
-            .afterStartup((servletContext) ->
-            {
-                JavaxWebSocketServerContainer serverContainer = JavaxWebSocketServerContainer.getContainer(servletContext);
-                if (configurator != null)
-                {
-                    try
-                    {
-                        configurator.accept(servletContext, serverContainer);
-                    }
-                    catch (DeploymentException e)
-                    {
-                        throw new RuntimeException("Failed to deploy WebSocket Endpoint", e);
-                    }
-                }
-            }));
+        context.addServletContainerInitializer(new JavaxWebSocketServletContainerInitializer(configurator));
```

Comment:
I don't understand why this is necessary? Why can't this ServletContainerInitializer be discovered by jetty in the same way as other ServletContainerInitializers?

Reply:
This method is only used for embedded Jetty. If you are running in a webapp it will be discovered with the ServiceLoader.